packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -369,7 +369,7 @@ IOSink? _openSink(String? output, {String basePath = ''}) {
   if (output == 'stdout') {
     sink = stdout;
   } else {
-    file = File(basePath + output);
+    file = File(path.join(basePath, output));
```

Comment:
Sorry, I just realized that we actually always want posix paths, because we're generating source, not dealing with paths on the generating platform; this will do the wrong thing on Windows.

We want all of these to be `path.posix.join`